ragatouille/RAGTrainer.py

```diff
@@ -94,9 +94,7 @@ def prepare_training_data(
         Returns:
             data_out_path: Union[str, Path] - Path to the directory where the data has been exported.
         """
-        if all_documents is not None:
-            self.collection += [doc for doc in all_documents if isinstance(doc, str)]
-
+        self.collection = (self.collection or []) + [doc['content'] for doc in all_documents]
```

Comment:
Same comment as below, I think the `'content'` key isn't actually an artifact we want to keep and build around in the long term, so ideally we'd keep the current logic and ensure it's been turned to string beforehand